src/Microsoft.ML.Transforms/GroupTransform.cs

```diff
@@ -263,10 +260,40 @@ public GroupSchema(Schema inputSchema, IHostEnvironment env, ModelLoadContext ct
 
                 KeepIds = GetColumnIds(inputSchema, _keepColumns, _ectx.Except);
 
-                _columnTypes = BuildColumnTypes(_input, KeepIds);
-                _columnNameMap = BuildColumnNameMap();
+                _columnTypes = BuildKeepColumnTypes(_input, KeepIds);
+
+                AsSchema = MakeOutputSchema();
+            }
 
-                AsSchema = Schema.Create(this);
+            /// <summary>
+            /// Create Schmea out of this object's attributes.
+            /// </summary>
+            private Schema MakeOutputSchema()
```

Comment:
We will have a lot of `MakeOutputSchema()` for each current implementation of `ISchema`. I am not sure if that is correct.